instrumentation/apache-httpasyncclient-4.0/src/main/java/io/opentelemetry/auto/instrumentation/apachehttpasyncclient/ApacheHttpAsyncClientTracer.java

```diff
@@ -78,8 +77,38 @@ protected String responseHeader(HttpResponse response, String name) {
     return header(response, name);
   }
 
+  @Override
+  protected Setter<HttpRequest> getSetter() {
+    return SETTER;
+  }
+
   private static String header(HttpMessage message, String name) {
     Header header = message.getFirstHeader(name);
     return header != null ? header.getValue() : null;
   }
+
+  @Override
+  protected String getInstrumentationName() {
+    return "io.opentelemetry.auto.apache-httpasyncclient-4.0";
+  }
+
+  /**
+   * This method is used to generate an acceptable CLIENT span (operation) name based on a given
+   * name.
+   */
+  @Override
+  public Span startSpan(String spanName) {
+    return tracer.spanBuilder(spanName).setSpanKind(Kind.CLIENT).startSpan();
+  }
+
+  @Override
+  public String spanNameForRequest(HttpRequest httpRequest) {
+    return super.spanNameForRequest(httpRequest);
+  }
+
+  /** This method is overridden to allow other classes in this package to call it. */
```

Comment:
The funny thing about protected methods is unless we make them `final` they may as well be `public` :P

Reply:
It's error prone to make it public in the super class.. user might call startSpan and then onRequest again.  in that case, onRequest will get invoked twice.  besides there are only a couple of places need to do this.